h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1227,14 +1229,50 @@ protected Frame rebalance(final Frame original_fr, boolean local, final String n
     return rebalanced_fr;
   }
 
+  private double rebalanceRatio() {
+    String mode = H2O.getCloudSize() == 1 ? "single" : "multi";
+    String ratioStr = System.getProperty(H2O.OptArgs.SYSTEM_PROP_PREFIX + "rebalance.ratio." + mode, "1.0");
+    return Double.parseDouble(ratioStr);
+  }
+
   /**
    * Find desired number of chunks. If fewer, dataset will be rebalanced.
    * @return Lower bound on number of chunks after rebalancing.
    */
   protected int desiredChunks(final Frame original_fr, boolean local) {
-    return Math.min((int) Math.ceil(original_fr.numRows() / 1e3), H2O.NUMCPUS);
+    if (H2O.getCloudSize() > 1 && Boolean.getBoolean(H2O.OptArgs.SYSTEM_PROP_PREFIX + "rebalance.enableMulti"))
+      return desiredChunk_multi(original_fr);
+    else
+      return desiredChunk_single(original_fr);
+  }
+
+  // single-node version (original version)
+  private int desiredChunk_single(final Frame originalFr) {
```

Comment:
Minor comment: the naming of method is weird - combines two name styles

Reply:
I will do just camelCase, I thought this would signify we have the 2 versions. But it does look weird